### PR TITLE
feat(hooks): implement <HierarchicalMenu>

### PR DESCRIPTION
**Pull request by dhayab**
# **Summary**

This introduces the `<HierarchicalMenu>` DOM widget.

Ticket: [FX-1152](https://algolia.atlassian.net/browse/FX-1152)

# **Implementation details**

Although the idea of having a common component for list handling has been evoked, after discussions, it seems like there are too few benefits of adding a level of depth in the relevant widgets.

# **Result**

- New `<HierarchicalMenu>` widget
- DOM tests for UI components and widget
- Replaced `<HierarchicalMenu>` with the widget in the hooks example

### CODE REVIEW

**Review by Haroenv — CHANGES_REQUESTED** [review]
show more can be implemented consistent with the others, and otherwise lgtm

File: examples/hooks/App.tsx [hunk 1]
```diff
@@ -85,6 +85,7 @@ export function App() {
                   'hierarchicalCategories.lvl2',
                 ]}
                 showMore={true}
+                limit={5}
```
[the remark this answers is not in the document]
Comment (dhayab) [reply examples/hooks/App.tsx]: I removed it in 8a1d2636.

File: packages/react-instantsearch-hooks-dom/src/ui/HierarchicalMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        props.className
+      )}
+    >
+      <HierarchicalList
+        classNames={classNames}
+        items={items}
+        onNavigate={onNavigate}
+        createURL={createURL}
+      />
+      {showMoreButton}
```
[the remark this answers is not in the document]
Comment (dhayab) [reply packages/react-instantsearch-hooks-dom/src/ui/HierarchicalMenu.tsx]: Indeed that's easier. I updated this in 8a1d2636.

**Review by Haroenv — APPROVED** [review]

File: packages/react-instantsearch-hooks-dom/src/ui/HierarchicalMenu.tsx [hunk 3]
```diff
@@ -7,35 +7,35 @@ import type { useHierarchicalMenu } from 'react-instantsearch-hooks';
...
-   * Class names to apply to any item element.
+   * Class names to apply to each item element
    */
   item: string;
   /**
-   * Class names to apply to a selected item element.
+   * Class names to apply to the selected item
    */
-  selectedItem: string;
+  itemSelected: string;
```
Comment (francoischalifour) [inline packages/react-instantsearch-hooks-dom/src/ui/HierarchicalMenu.tsx:29]: Why this change? It's `selectedItem` in InstantSearch.js.

**Review by FabienMotte — APPROVED** [review]
Nice work!

File: packages/react-instantsearch-hooks-dom/src/ui/HierarchicalMenu.tsx (hunk 3 above)
Comment (dhayab) [reply packages/react-instantsearch-hooks-dom/src/ui/HierarchicalMenu.tsx:29]: I checked the other Hooks DOM components and made the change for consistency: https://github.com/algolia/react-instantsearch/search?q=itemSelected.

Comment (francoischalifour) [reply packages/react-instantsearch-hooks-dom/src/ui/HierarchicalMenu.tsx:29]: We should aim for consistency with IS.js imo.

Comment (dhayab) [reply packages/react-instantsearch-hooks-dom/src/ui/HierarchicalMenu.tsx:29]: You're right. I created a [task](https://algolia.atlassian.net/browse/FX-1324) to list all the adjustments we should make on the components before releasing them.
